kratos/processes/find_intersected_geometrical_objects_process.cpp

```diff
@@ -147,9 +147,12 @@ namespace Kratos
 			}
 		}
 
-    for(int i = 0 ; i < 3; i++){
-			low[i] -= std::numeric_limits<double>::epsilon();
-			high[i] += std::numeric_limits<double>::epsilon();
+		// Slightly increase the bounding box size to avoid problems with geometries in the borders
+		// Note that std::numeric_limits<double>::double() is added for the 2D cases. Otherwise, the
+		// third component will be 0, breaking the octree behaviour.
+    		for(int i = 0 ; i < 3; i++) {
+			low[i] -= std::abs(high[i] - low[i])*1e-3 + std::numeric_limits<double>::epsilon();
```

Comment:
It would be better to use a `constexpr double relative_epsilon = 1e-3;` to be more clear the code and easier to be changed in the future

Reply:
Do you want me to change it in a fast PR or can we leave it for the future?

Reply:
I would put a fast PR